src/lib/components/healthselectorv2/HealthSelector.component.js

```diff
@@ -0,0 +1,233 @@
+  ${CSSCircle}:first-child {
+    margin-right: -20%;
+  }
+  ${CSSCircle}:last-child {
+    margin-left: -20%;
+  }
+  height: 16px;
+`;
+
+const ValueIcon = styled.span`
+  margin: 0 ${defaultTheme.spacing.sp8};
+  height: 100%;
+  width: 100%;
+`;
+
+const OptionIcon = styled.span`
+  width: 35px;
+  margin-right: ${defaultTheme.spacing.sp4};
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const OptionWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  height: 100%;
+  justify-content: flex-start;
+  overflow: hidden;
+  line-height: 18px;
+`;
+
+const OptionLabel = styled.span`
+  display: flex;
+  align-items: bottom;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  display: block;
+}
+`;
+
+const SelectStyle = styled(Select)`
+  .sc-select__control {
+    background-color: ${defaultTheme.defaultTheme.darkRebrand.buttonSecondary};
+
+    &.sc-select__control--is-focused {
+      border: 2px dashed ${defaultTheme.defaultTheme.darkRebrand.selectedActive};
+      background-color: ${defaultTheme.defaultTheme.darkRebrand
+        .buttonSecondary};
+    }
+    .sc-select__value-container {
+      overflow: visible;
+    }
+
+    &.sc-select__control--menu-is-open {
+      .sc-select__indicator {
+        transform: rotate(180deg);
+      }
+    }
+  }
+`;
+
+const icons = {
+  all: (
+    <ThreeCirclesWrapper>
+      <CSSCircle variant="statusHealthy" />
+      <CSSCircle variant="statusWarning" />
+      <CSSCircle variant="statusCritical" />
+    </ThreeCirclesWrapper>
+  ),
+  healthy: <Icon name={'Check-circle'} color={'statusHealthy'} size={'lg'} />,
+  warning: (
+    <Icon name={'Exclamation-circle'} color={'statusWarning'} size={'lg'} />
+  ),
+  critical: <Icon name={'Times-circle'} color={'statusCritical'} size={'lg'} />,
+  unknown: <Icon name={'Info'} color={'infoPrimary'} size={'lg'} />,
+};
+
+const optionsDefaultConfiguration: optionsConfiguration = {
+  all: {
+    icon: icons.all,
+    label: 'All healthy statuses',
+    shortLabel: 'All',
+    value: 'all',
+    isHidden: false,
+  },
+  healthy: {
+    icon: icons.healthy,
+    label: 'Healthy only',
+    shortLabel: 'Healthy',
+    value: 'healthy',
+    isHidden: false,
+  },
+  critical: {
+    icon: icons.critical,
+    label: 'Critical only',
+    shortLabel: 'Critical',
+    value: 'critical',
+    isHidden: false,
+  },
+  warning: {
+    icon: icons.warning,
+    label: 'Warning only',
+    shortLabel: 'Warning',
+    value: 'warning',
+    isHidden: false,
+  },
+  unknown: {
+    icon: icons.unknown,
+    label: 'Unknown only',
+    shortLabel: 'Unknown',
+    value: 'unknown',
+    disabled: false,
+  },
+};
+
+function HealthSelectorv2(props: Props) {
+  const {
+    optionsConfiguration,
+    label = 'Health',
+    onChange,
+    defaultValue = 'all',
+    ...rest
+  } = props;
+
+  const options = [];
+
+  Object.keys({
+    ...optionsDefaultConfiguration,
+    ...optionsConfiguration,
```

Comment:
instead of merging these config I would just pass a bunch of optionsConfig as a prop and use that prop directly here to ease composability. By default if no options is passed we use the whole default ones.